### PR TITLE
Support PHP7.1 nullable types

### LINKED ISSUE #290: Exception when using nullable types
_From @androlets on February 8, 2017 16:3_

After latest extension update (v1.1.1) a get this error when i hover over methods
```
Message: ErrorException: Object of class PhpParser\Node\NullableType could not be converted to string in .vscode/extensions/felixfbecker.php-intellisense-1.1.1/vendor/felixfbecker/language-server/src/DefinitionResolver.php:759
```

```
#0 /home/andrejs/.vscode/extensions/felixfbecker.php-intellisense-1.1.1/vendor/felixfbecker/language-server/src/DefinitionResolver.php(759): {closure}(4096, 'Object of class...', '/home/andrejs/....', 759, Array)
#1 /home/andrejs/.vscode/extensions/felixfbecker.php-intellisense-1.1.1/vendor/felixfbecker/language-server/src/DefinitionResolver.php(131): LanguageServer\DefinitionResolver->getTypeFromNode(Object(PhpParser\Node\Stmt\ClassMethod))
#2 /home/andrejs/.vscode/extensions/felixfbecker.php-intellisense-1.1.1/vendor/felixfbecker/language-server/src/NodeVisitor/DefinitionCollector.php(45): LanguageServer\DefinitionResolver->createDefinitionFromNode(Object(PhpParser\Node\Stmt\ClassMethod), 'Skus\\Zone->tari...')
#3 /home/andrejs/.vscode/extensions/felixfbecker.php-intellisense-1.1.1/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php(159): LanguageServer\NodeVisitor\DefinitionCollector->enterNode(Object(PhpParser\Node\Stmt\ClassMethod))
#4 /home/andrejs/.vscode/extensions/felixfbecker.php-intellisense-1.1.1/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php(101): PhpParser\NodeTraverser->traverseArray(Array)
#5 /home/andrejs/.vscode/extensions/felixfbecker.php-intellisense-1.1.1/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php(171): PhpParser\NodeTraverser->traverseNode(Object(PhpParser\Node\Stmt\Class_))
#6 /home/andrejs/.vscode/extensions/felixfbecker.php-intellisense-1.1.1/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php(101): PhpParser\NodeTraverser->traverseArray(Array)
#7 /home/andrejs/.vscode/extensions/felixfbecker.php-intellisense-1.1.1/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php(171): PhpParser\NodeTraverser->traverseNode(Object(PhpParser\Node\Stmt\Namespace_))
#8 /home/andrejs/.vscode/extensions/felixfbecker.php-intellisense-1.1.1/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php(85): PhpParser\NodeTraverser->traverseArray(Array)
#9 /home/andrejs/.vscode/extensions/felixfbecker.php-intellisense-1.1.1/vendor/felixfbecker/language-server/src/PhpDocument.php(205): PhpParser\NodeTraverser->traverse(Array)
#10 /home/andrejs/.vscode/extensions/felixfbecker.php-intellisense-1.1.1/vendor/felixfbecker/language-server/src/PhpDocument.php(119): LanguageServer\PhpDocument->updateContent('<?php\ndeclare(s...')
#11 /home/andrejs/.vscode/extensions/felixfbecker.php-intellisense-1.1.1/vendor/felixfbecker/language-server/src/PhpDocumentLoader.php(134): LanguageServer\PhpDocument->__construct('file:///home/an...', '<?php\ndeclare(s...', Object(LanguageServer\Index\Index), Object(LanguageServer\Parser), Object(phpDocumentor\Reflection\DocBlockFactory), Object(LanguageServer\DefinitionResolver))
#12 /home/andrejs/.vscode/extensions/felixfbecker.php-intellisense-1.1.1/vendor/felixfbecker/language-server/src/PhpDocumentLoader.php(113): LanguageServer\PhpDocumentLoader->create('file:///home/an...', '<?php\ndeclare(s...')
#13 [internal function]: LanguageServer\PhpDocumentLoader->LanguageServer\{closure}()
#14 /home/andrejs/.vscode/extensions/felixfbecker.php-intellisense-1.1.1/vendor/sabre/event/lib/coroutine.php(70): Generator->send('<?php\ndeclare(s...')
#15 /home/andrejs/.vscode/extensions/felixfbecker.php-intellisense-1.1.1/vendor/sabre/event/lib/Promise.php(242): Sabre\Event\{closure}('<?php\ndeclare(s...')
#16 /home/andrejs/.vscode/extensions/felixfbecker.php-intellisense-1.1.1/vendor/sabre/event/lib/Loop/Loop.php(261): Sabre\Event\Promise->Sabre\Event\{closure}()
#17 /home/andrejs/.vscode/extensions/felixfbecker.php-intellisense-1.1.1/vendor/sabre/event/lib/Loop/Loop.php(215): Sabre\Event\Loop\Loop->runNextTicks()
#18 /home/andrejs/.vscode/extensions/felixfbecker.php-intellisense-1.1.1/vendor/sabre/event/lib/Loop/Loop.php(194): Sabre\Event\Loop\Loop->tick(true)
#19 /home/andrejs/.vscode/extensions/felixfbecker.php-intellisense-1.1.1/vendor/sabre/event/lib/Loop/functions.php(122): Sabre\Event\Loop\Loop->run()
#20 /home/andrejs/.vscode/extensions/felixfbecker.php-intellisense-1.1.1/vendor/felixfbecker/language-server/bin/php-language-server.php(102): Sabre\Event\Loop\run()
#21 {main}
```

_Copied from original issue: felixfbecker/vscode-php-intellisense#69_

### PR DESCRIPTION
**Pull request by felixfbecker**
Closes #290

### CODE REVIEW

File: src/DefinitionResolver.php [hunk 1]
```diff
@@ -701,11 +701,21 @@ private static function resolveClassNameToType(Node $class): Type
...
+     * @param Node|string $node
      * @return \phpDocumentor\Type|null
      */
-    public function getTypeFromNode(Node $node)
+    public function getTypeFromNode($node)
     {
+        if (is_string($node)) {
+            // Resolve a string like "bool" to a type object
+            return $this->typeResolver->resolve($node->returnType);
+        }
```
Comment (jrobeson) [inline src/DefinitionResolver.php]: this can't work if $node is a string.

Comment (felixfbecker) [inline src/DefinitionResolver.php]: woops, this is meant to read `resolve($node)`

**Comment by jrobeson** [comment]
I applied this patch locally (with the string case fix) and it seemed to work.

**Comment by felixfbecker** [comment]
Cool, thanks for trying out! Tests are failing though, this seems to have caused some side-effect

**Comment by jrobeson** [comment]
yeah, I cared more about getting rid of the problems window constantly popping up than I did about making sure it properly worked :(

**Comment by jens1o** [comment]
any news on this?

**Comment by jens1o** [comment]
cursed...

**Comment by jrobeson** [comment]
this can be closed now, right?

**Pull request closed without merging**